### PR TITLE
Update Belarus holidays: add ru & th l10n, 2024-2025 substituted holidays

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Implements the following changes for Belarus holidays:
- Adds Russian (`ru`) and Thai (`th`) localization - the former based on [the official Belarus government source]( https://president.gov.by/ru/gosudarstvo/prazdniki).
- Adds substituted holidays for 2024 and 2025.
- Adds `WORKDAY` category support for those considered as [National/Country-wide/Religious holidays](https://president.gov.by/en/gosudarstvo/prazdniki) but don't belong in [the current Non-Working day status](https://president.gov.by/en/gosudarstvo/prazdniki/calendar-2024).
- Extends support start date down to 1991.
- Refractors testcase to better match modern library-wide standards.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

**Comment by Prateekshit73** [comment]
Hello @PPsyrius,

I hope this message finds you well. **I apologize for the interruption and for bringing up a different topic**, but I am currently facing an issue with the testing portion of the holidays project, specifically related to the Afghanistan holidays implementation.

While running the tests, I encountered an assertion error indicating a mismatch between the expected and actual supported languages for Afghanistan. The test expects only ['fa'], but the implementation supports both ['fa', 'ps']. I want to ensure that I am approaching this correctly and would greatly appreciate any guidance on how to resolve this discrepancy.

I attempted to create a new discussion related to this issue, but unfortunately, I was unable to perform that action.

If there are any specific steps or considerations I should keep in mind while addressing this issue, I would be very grateful for your insights.

Thank you for your time and assistance!

**Comment by PPsyrius** [comment]
> Hello @PPsyrius,
> 
> I hope this message finds you well. **I apologize for the interruption and for bringing up a different topic**, but I am currently facing an issue with the testing portion of the holidays project, specifically related to the Afghanistan holidays implementation.
> 
> While running the tests, I encountered an assertion error indicating a mismatch between the expected and actual supported languages for Afghanistan. The test expects only ['fa'], but the implementation supports both ['fa', 'ps']. I want to ensure that I am approaching this correctly and would greatly appreciate any guidance on how to resolve this discrepancy.
> 
> I attempted to create a new discussion related to this issue, but unfortunately, I was unable to perform that action.
> 
> If there are any specific steps or considerations I should keep in mind while addressing this issue, I would be very grateful for your insights.
> 
> Thank you for your time and assistance!

No worries, have you added the `supported_languages` entry for your own `holidays\countries\afghanistan.py` file yet? 

Not to be confused with `default_language` here.

**Comment by Prateekshit73** [comment]
> > Hello @PPsyrius,
> > I hope this message finds you well. **I apologize for the interruption and for bringing up a different topic**, but I am currently facing an issue with the testing portion of the holidays project, specifically related to the Afghanistan holidays implementation.
> > While running the tests, I encountered an assertion error indicating a mismatch between the expected and actual supported languages for Afghanistan. The test expects only ['fa'], but the implementation supports both ['fa', 'ps']. I want to ensure that I am approaching this correctly and would greatly appreciate any guidance on how to resolve this discrepancy.
> > I attempted to create a new discussion related to this issue, but unfortunately, I was unable to perform that action.
> > If there are any specific steps or considerations I should keep in mind while addressing this issue, I would be very grateful for your insights.
> > Thank you for your time and assistance!
> 
> No worries, have you added the `supported_languages` entry for your own `holidays\countries\afghanistan.py` file yet?
> 
> Not to be confused with `default_language` here.

Yes Sir
``` 
country = "AF"
default_language = "fa"
supported_languages = ("fa", "ps")
```

File: holidays/countries/belarus.py [hunk 1]
```diff
@@ -43,36 +50,108 @@ def __init__(self, *args, **kwargs):
...
+
+        # Radunitsa (Day of Rejoicing).
+        self._add_rejoicing_day(tr("Радаўніца"))
 
         # Catholic Christmas Day.
         self._add_christmas_day(tr("Нараджэнне Хрыстова (каталіцкае Раство)"), GREGORIAN_CALENDAR)
 
+        if 1992 <= self._year <= 1997:
+            # Catholic Easter.
+            name = tr("каталiцкi Вялiкдзень")
```
Comment (KJhellico) [inline holidays/countries/belarus.py:100]:
```suggestion
            name = tr("Каталiцкi Вялiкдзень")
```

File: holidays/countries/belarus.py [hunk 2]
```diff
@@ -43,36 +50,108 @@ def __init__(self, *args, **kwargs):
...
+            self._add_easter_sunday(name, GREGORIAN_CALENDAR)
+            self._add_easter_monday(name, GREGORIAN_CALENDAR)
+
+            # Orthodox Easter.
+            name = tr("праваслаўны Вялiкдзень")
```
Comment (KJhellico) [inline holidays/countries/belarus.py:105]:
```suggestion
            name = tr("Праваслаўны Вялiкдзень")
```

File: holidays/countries/belarus.py [hunk 3]
```diff
@@ -43,36 +50,108 @@ def __init__(self, *args, **kwargs):
...
+        # Country-wide holidays
+
+        # Day of the Fatherland's Defenders and the Armed Forces of the Republic of Belarus.
+        self._add_holiday_feb_23(tr("Дзень абаронцаў Айчыны і Узброеных Сіл Рэспублікі Беларусь"))
+
+        # Religious holidays
+
+        if self._year >= 1998:
+            # Catholic Easter.
+            self._add_easter_sunday(tr("каталiцкi Вялiкдзень"), GREGORIAN_CALENDAR)
```
Comment (KJhellico) [inline holidays/countries/belarus.py:147]:
```suggestion
            self._add_easter_sunday(tr("Каталiцкi Вялiкдзень"), GREGORIAN_CALENDAR)
```

File: holidays/countries/belarus.py [hunk 4]
```diff
@@ -43,36 +50,108 @@ def __init__(self, *args, **kwargs):
...
+
+            # Orthodox Easter.
+            self._add_easter_sunday(tr("праваслаўны Вялiкдзень"))
```
Comment (KJhellico) [inline holidays/countries/belarus.py:150]:
```suggestion
            self._add_easter_sunday(tr("Праваслаўны Вялiкдзень"))
```

File: holidays/locale/ru/LC_MESSAGES/BY.po [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+msgid "Нараджэнне Хрыстова (праваслаўнае Раство)"
+msgstr "Рождество Христово (православное Рождество)"
+
+#. Catholic Easter.
+msgid "каталiцкi Вялiкдзень"
+msgstr "Католическая Пасха"
+
+#. Orthodox Easter.
+msgid "праваслаўны Вялiкдзень"
+msgstr "Пасха православная"
```
Comment (KJhellico) [inline holidays/locale/ru/LC_MESSAGES/BY.po:99]:
```suggestion
msgstr "Православная Пасха"
```

File: tests/countries/test_belarus.py [hunk 6]
```diff
@@ -25,37 +26,181 @@ def test_country_aliases(self):
...
-            "2020-01-01",
-            "2020-01-02",
-            "2021-01-01",
-            "2021-01-02",
+    def test_constitution_day(self):
+        name = "Дзень Канстытуцыі"
+
+        workdays = Belarus(categories=WORKDAY, years=range(1991, 2050))
+        self.assertHolidayName(name, workdays, (f"{year}-03-15" for year in range(1999, 2050)))
+        self.assertNoHolidayName(name, workdays, (f"{year}-03-15" for year in range(1991, 1995)))
```
Comment (KJhellico) [inline tests/countries/test_belarus.py:59]:
```suggestion
        self.assertNoHolidayName(name, workdays, range(1991, 1995))
```
More strict check.

File: tests/countries/test_belarus.py [hunk 7]
```diff
@@ -25,37 +26,181 @@ def test_country_aliases(self):
...
+            (f"{year}-03-15" for year in range(1991, 1995)),
+            (f"{year}-03-15" for year in range(1999, 2050)),
+        )
+
+    def test_day_of_unity_of_the_peoples_of_belarus_and_russia(self):
+        name = "Дзень яднання народаў Беларусі і Расіі"
+
+        workdays = Belarus(categories=WORKDAY, years=range(1991, 2050))
+        self.assertHolidayName(name, workdays, (f"{year}-04-02" for year in range(1996, 2050)))
+        self.assertNoHolidayName(name, workdays, (f"{year}-04-02" for year in range(1991, 1996)))
```
Comment (KJhellico) [inline tests/countries/test_belarus.py:72]:
```suggestion
        self.assertNoHolidayName(name, workdays, range(1991, 1996))
```

File: tests/countries/test_belarus.py [hunk 8]
```diff
@@ -25,37 +26,181 @@ def test_country_aliases(self):
...
+        self.assertHolidayName(name, (f"{year}-07-03" for year in range(1997, 2050)))
+        self.assertNoHolidayName(name, (f"{year}-07-27" for year in range(1997, 2050)))
+        self.assertNoHolidayName(name, (f"{year}-07-03" for year in range(1991, 1997)))
+
+    def test_day_of_peoples_unity(self):
+        name = "Дзень народнага адзінства"
+
+        workdays = Belarus(categories=WORKDAY, years=range(1991, 2050))
+        self.assertHolidayName(name, workdays, (f"{year}-09-17" for year in range(2021, 2050)))
+        self.assertNoHolidayName(name, workdays, (f"{year}-09-17" for year in range(1991, 2021)))
```
Comment (KJhellico) [inline tests/countries/test_belarus.py:117]:
```suggestion
        self.assertNoHolidayName(name, workdays, range(1991, 2021))
```

**Comment by arkid15r** [comment]
> > > Hello @PPsyrius,
> > > I hope this message finds you well. **I apologize for the interruption and for bringing up a different topic**, but I am currently facing an issue with the testing portion of the holidays project, specifically related to the Afghanistan holidays implementation.
> > > While running the tests, I encountered an assertion error indicating a mismatch between the expected and actual supported languages for Afghanistan. The test expects only ['fa'], but the implementation supports both ['fa', 'ps']. I want to ensure that I am approaching this correctly and would greatly appreciate any guidance on how to resolve this discrepancy.
> > > I attempted to create a new discussion related to this issue, but unfortunately, I was unable to perform that action.
> > > If there are any specific steps or considerations I should keep in mind while addressing this issue, I would be very grateful for your insights.
> > > Thank you for your time and assistance!
> > 
> > 
> > No worries, have you added the `supported_languages` entry for your own `holidays\countries\afghanistan.py` file yet?
> > Not to be confused with `default_language` here.
> 
> Yes Sir
> 
> ```
> country = "AF"
> default_language = "fa"
> supported_languages = ("fa", "ps")
> ```

@Prateekshit73 you can use https://github.com/vacanza/holidays/issues/1144 for discussion. 
Also make sure that README.rst file describes the supported languages correctly.

File: tests/countries/test_belarus.py [hunk 9]
```diff
@@ -25,37 +26,181 @@ def test_country_aliases(self):
...
+
+        self.assertHolidayName(name, (f"{year}-07-27" for year in range(1991, 1997)))
+        self.assertHolidayName(name, (f"{year}-07-03" for year in range(1997, 2050)))
+        self.assertNoHolidayName(name, (f"{year}-07-27" for year in range(1997, 2050)))
+        self.assertNoHolidayName(name, (f"{year}-07-03" for year in range(1991, 1997)))
+
+    def test_day_of_peoples_unity(self):
+        name = "Дзень народнага адзінства"
+
+        workdays = Belarus(categories=WORKDAY, years=range(1991, 2050))
```
Comment (KJhellico) [inline tests/countries/test_belarus.py:115]: I suggest to add
```python
    cls.workday_holidays = Belarus(categories=WORKDAY, years=range(1991, 2050))
```
into `setUpClass(cls)` and use it when needed (as `self.workday_holidays`).

File: holidays/countries/belarus.py [hunk 10]
```diff
@@ -43,36 +50,108 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def _populate_public_holidays(self):
-        # New Year's Day.
-        self._add_new_years_day(tr("Новы год"))
+        # National holidays
```
Comment (KJhellico) [inline holidays/countries/belarus.py:53]: I think it's more convenient to keep chronological holidays order.

**Review by KJhellico — DISMISSED** [review]
LGTM.

File: holidays/countries/belarus.py [hunk 11]
```diff
@@ -44,35 +51,95 @@ def __init__(self, *args, **kwargs):
...
+        if 1992 <= self._year <= 1997:
+            # Catholic Easter.
+            name = tr("Каталiцкi Вялiкдзень")
+            self._add_easter_sunday(name, GREGORIAN_CALENDAR)
+            self._add_easter_monday(name, GREGORIAN_CALENDAR)
+
+            # Orthodox Easter.
+            name = tr("Праваслаўны Вялiкдзень")
+            self._add_easter_sunday(name)
+            self._add_easter_monday(name)
```
Comment (arkid15r) [inline holidays/countries/belarus.py:101]: @PPsyrius  @KJhellico could you double check this part?
It seems the 1992-1997 range is [applicable](https://ru.wikipedia.org/wiki/%D0%9F%D1%80%D0%B0%D0%B7%D0%B4%D0%BD%D0%B8%D0%BA%D0%B8_%D0%91%D0%B5%D0%BB%D0%BE%D1%80%D1%83%D1%81%D1%81%D0%B8%D0%B8#%D0%A0%D0%B5%D0%BB%D0%B8%D0%B3%D0%B8%D0%BE%D0%B7%D0%BD%D1%8B%D0%B5) to Monday after Easter Sunday only.

Should it also have a different name (or observed suffix), WDYT?

**Review by arkid15r — COMMENTED** [review]
LGTM,

just want to make sure the Easter Monday is implemented correctly:

File: holidays/countries/belarus.py (hunk 11 above)
Comment (PPsyrius) [reply holidays/countries/belarus.py:101]: @KJhellico @arkid15r I'll move Belarus's Easter to the main public list as default then 👍

**Review by KJhellico — APPROVED** [review]
LGTM.

**Review by arkid15r — COMMENTED** [review]
@KJhellico @PPsyrius thanks for taking care of that!

LGTM

**Pull request merged**